tests/v2/actions/etcdsnapshot/creates.go

```diff
@@ -0,0 +1,709 @@
+		if err == nil {
+			return errors.New("expecting cluster restore to remove resource")
+		}
+
+		_, err = steveclient.SteveType(serviceType).ByID(postServiceResp.ID)
+		if err == nil {
+			return errors.New("expecting cluster restore to remove resource")
+		}
+
+	} else {
+		cluster, snapshotName, postDeploymentResp, postServiceResp, err := CreateAndValidateSnapshotV2Prov(client, podTemplate, deploymentTemplate, clusterName, clusterID, etcdRestore, isRKE1)
+		if err != nil {
+			return err
+		}
+
+		err = RestoreAndValidateSnapshotV2Prov(client, snapshotName, etcdRestore, cluster, clusterID)
+		if err != nil {
+			return err
+		}
+
+		_, err = steveclient.SteveType(DeploymentSteveType).ByID(postDeploymentResp.ID)
+		if err == nil {
+			return errors.New("expecting cluster restore to remove resource")
+		}
+
+		_, err = steveclient.SteveType(serviceType).ByID(postServiceResp.ID)
+		if err == nil {
+			return errors.New("expecting cluster restore to remove resource")
+		}
+	}
+
+	logrus.Infof("Deleting created workloads...")
+	err = steveclient.SteveType(DeploymentSteveType).Delete(deploymentResp)
+	if err != nil {
+		return err
+	}
+
+	err = steveclient.SteveType(serviceType).Delete(serviceResp)
+	if err != nil {
+		return err
+	}
+
+	err = steveclient.SteveType(ingressSteveType).Delete(ingressResp)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+// CreateAndValidateSnapshotRKE1 is a helper that takes a snapshot of a given rke1 cluster and validates is resources after the snapshot
+func CreateAndValidateSnapshotRKE1(client *rancher.Client, podTemplate *corev1.PodTemplateSpec, deployment *v1.Deployment, clusterName, clusterID string,
+	etcdRestore *Config, isRKE1 bool) (*management.Cluster, string, *steveV1.SteveAPIObject, *steveV1.SteveAPIObject, error) {
+
+	createdSnapshots, err := shepherdsnapshot.CreateRKE1Snapshot(client, clusterName)
+	if err != nil {
+		return nil, "", nil, nil, err
+	}
+
+	cluster, err := client.Management.Cluster.ByID(clusterID)
+	if err != nil {
+		return nil, "", nil, nil, err
+	}
+
+	if etcdRestore.ReplaceRoles != nil && cluster.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig.S3BackupConfig != nil {
+		// placeholder for now. May have to refactore scaling
+		err = scalinginput.ReplaceRKE1Nodes(client, clusterName, etcdRestore.ReplaceRoles.Etcd, etcdRestore.ReplaceRoles.ControlPlane, etcdRestore.ReplaceRoles.Worker)
+		if err != nil {
+			return nil, "", nil, nil, err
+		}
+	}
+
+	snapshotToRestore := createdSnapshots[0].ID
+	createdSnapshotIDs := []string{}
+	isSnapshotS3 := false
+
+	// prioritize s3 snapshots over local.
+	for _, snapshot := range createdSnapshots {
+		if snapshot.BackupConfig.S3BackupConfig != nil {
+			snapshotToRestore = snapshot.ID
+			isSnapshotS3 = true
+		}
+		createdSnapshotIDs = append(createdSnapshotIDs, snapshot.ID)
+	}
+	if cluster.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig.S3BackupConfig != nil && !isSnapshotS3 {
```

Comment:
This is interesting, so in essence, if S3 is enabled but a local snapshot is selected, the test is going to fail here? Is this something we really want to be doing? It almost seems that if this is the case, we should enhance the logic which I think you already did.